sklearn/preprocessing/tests/test_data.py

```diff
@@ -905,6 +951,20 @@ def test_robust_scaler_2d_arrays():
     assert_array_almost_equal(X_scaled.std(axis=0)[0], 0)
 
 
+def test_robust_scaler_equivalence_dense_sparse():
+    # Check the equivalence of the fitting with dense and sparse matrices
+    X_sparse = sparse.rand(1000, 5, density=0.5).tocsc()
```

Comment:
To avoid regressions, please make sure this works with an all-positive, all-negative, all-zero, positive, negative and zero, and all-nonzero columns. Check with a couple of densities and also integer dtypes.

Reply:
NaNs with integer dtype should not work.